src/ensembl/production/metadata/api/factories/release.py

```diff
@@ -86,11 +86,21 @@ def init_release(
                 version = round(version, 1)
 
             # Validate release date only if provided
+            release_date_obj = None
             if release_date:
                 try:
-                    datetime.strptime(release_date, "%Y-%m-%d").date()
+                    release_date_obj = datetime.strptime(release_date, "%Y-%m-%d").date()
                 except ValueError:
                     raise ValueError("Invalid release_date format. Expected YYYY-MM-DD.")
+            else:
+                if label:
+                    try:
+                        release_date_obj = datetime.strptime(label, "%Y-%m-%d").date()
+                        release_date = label  # Store the string for later label assignment
+                    except ValueError:
+                        raise ValueError("Invalid label format. Expected YYYY-MM-DD when used as date.")
+                else:
+                    raise ValueError("Either release_date or label must be specified.")
```

Comment:
In terms of efficiency, I would move the check of "Either release_date or label must be specified." before you do any database connection - but it is not critical.

```suggestion
                if label is None:
                    label = release_date
            elif label:
                try:
                    release_date_obj = datetime.strptime(label, "%Y-%m-%d").date()
                except ValueError:
                    raise ValueError("Invalid label format. Expected YYYY-MM-DD when used as date.")
            else:
                raise ValueError("Either 'release_date' or 'label' must be specified.")
```
`release_date` is not used afterwards. I would also raise the fact that `label = ""` would be left as it is in this case if `release_date` is also provided (not sure if this is a wanted behaviour).